### PR TITLE
Create multitenancy.md

### PR DESCRIPTION
**Pull request by sergeyborovoy**
If the "What it is" section looks too verbose, the last 1 or even 2 paragraphs of it can be deleted, although I have re-read everything carefully and do believe that they provide important supplementary information that is not redundant. 

I confirm that this is completely my own work, although in the world of term definitions you can always find fragments similar to each other even if they were developed independently.

Signed-off-by: Sergey Borovoy <48699431+sergeyborovoy@users.noreply.github.com>

### CODE REVIEW

**Comment by sergeyborovoy** [comment]
The spellchecker says that multitenancy is not a word.  Apparently, it prefers "multi-tenancy".  Well, it's spelt without the dash in sources like Wikipedia, Gartner, RedHat, SAP, Google (https://cloud.google.com/datastore/docs/concepts/multitenancy) and many more.  (Spelling with the dash is also common, which I pointed out in the article. ) 

I believe that the spellchecker should be overridden.

**Comment by seokho-son** [comment]
Hi @sergeyborovoy 
You can add `multitenancy` (also related new words if necessary) in https://github.com/cncf/glossary/blob/main/wordlist.txt
The commit for updated wordlist.txt also can be included in this PR. :)

**Comment by sergeyborovoy** [comment]
Sorry I'm not a fluent github user so the best I could do is a new PR :)

**Review by iamNoah1 — COMMENTED** [review]
I'd say come to slack and we will help you update the wordlist in this PR as well. I guess it is better to have both in the same PR. What do you think @seokho-son, @jihoon-seo ?

**Comment by seokho-son** [comment]
Hi @iamNoah1 :)

> it is better to have both in the same PR

I agree with you. But, since @sergeyborovoy already created a PR #1157 to update wordlist, let's proceed ..

@sergeyborovoy when you define a new term next time, please try to add updated wordlist with in the same PR. :)

**Comment by jihoon-seo** [comment]
force-push: added the word `multitenant` to `wordlist.txt`, which is also needed to pass spellcheck but not covered in the PR #1157.

**Review by CathPag — COMMENTED** [review]
This is great, @sergeyborovoy! I haven't finished reviewing yet but did the first section. I'm the editor reviewer and can't provide much feedback on technical accuracy, but it all sounds pretty accurate :)

Most of my requested changes are edits, but I also have one addition for clarification and feedback on the direction of the 2nd section. I'll get to the rest of it early next week.

File: content/en/multitenancy.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
+title: Multitenancy
+status: Feedback Appreciated
+category: Property
+---
+
+## What it is
+
+Multitenancy (often written as multi-tenancy) refers to the capability of a single software installation to serve multiple tenants.  A tenant in this context is a user, application, or a group of users / applications that use the software to operate on their own set of data, and this data is only shared with other tenants as explicitly instructed by the owner – or, more often, not at all.
```
[the remark this answers is not in the document]
Comment (sergeyborovoy) [reply content/en/multitenancy.md:9]: I guess the word "capability" here means that multitenant software may also serve just one tenant (let's say you installed it on-prem just for yourself...) but if it **_capable_** of serving multiple tenants, then it's already multitenant.

Tenants are normally not aware of each other, but they may be if they choose to be discoverable.   Say, GitHub is multitenant, but its whole point is enabing the tenants to cooperate.

File: content/en/multitenancy.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+title: Multitenancy
+status: Feedback Appreciated
+category: Property
+---
+
+## What it is
+
+Multitenancy (often written as multi-tenancy) refers to the capability of a single software installation to serve multiple tenants.  A tenant in this context is a user, application, or a group of users / applications that use the software to operate on their own set of data, and this data is only shared with other tenants as explicitly instructed by the owner – or, more often, not at all.  
+
+In other words, multitenant software provides each tenant with their own segregated environment (work data, settings, list of credentials etc), serving multiple tenants at the same time and in a way that for each tenant it looks and feels as if they use their own dedicated installation of the software.  To achieve that, the software runs on a server and provides a way for the tenants to connect to it over the network, offering them a user interface and/or an [API](https://glossary.cncf.io/application-programming-interface) (see [Client-Server Architecture](https://glossary.cncf.io/client-server-architecture)).
```
[the remark this answers is not in the document]
Comment (sergeyborovoy) [reply content/en/multitenancy.md:11]: I agree

File: content/en/multitenancy.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+Typical and well-known examples of multitenant software are Google Mail, Google Docs, Microsoft Office 365, Salesforce CRM, Dropbox, with thousands more falling into this category fully or at least partially.
+
+A tenant can be as small as just one user with a single login ID, completely independent of other users (as it is often the case with personal productivity software), or as large as an entire corporation with thousands of logins IDs each having their own privileges but interrelated in a multitude of ways. 
+
+## Problem it addresses 
+
+Multitenancy is a key to operational efficiency of server-run software.  Providing each tenant with a dedicated installation is wasteful in terms of hardware utilization and maintenance efforts.
```
[the remark this answers is not in the document]
Comment (sergeyborovoy) [reply content/en/multitenancy.md:19]: I guess the first phrase can be moved to solution, yes.

**Comment by CathPag** [comment]
@sergeyborovoy, looks like you closed comments and completely ignored my edits. You can still accept them and change whatever you think I might have gotten wrong and **start a conversation** (explain why you disagree). But please don't simply disregard feedback from Glossary maintainers. There are some stylistic improvements I would rather not lose.

**Comment by seokho-son** [comment]
Hi @sergeyborovoy 
I think you don't have to click `Resolve conversation` 
unless you are completely sure all folks involved in a review conversation are on a same page. (if a maintainer is involved, you'd better to wait until the maintainer agree with the result of a conversation. (because maintainers are highly responsible of all contents as well as the project itself)
I guess you are not familiar with Github and interactions in an opensource community yet. Please consider this is a learning curve.
Could you reopen closed conversations and re-initiate conversation ? (you can click `Unresolve conversation` button.

**Comment by sergeyborovoy** [comment]
Hi @CathPag, hi @seokho-son 
My apologies, I guess was under the impression that by clicking "Resolve conversation", I tell github to merge the suggested changes.  Like I said, I'm not a regular user of github.
I just unresolved everything and clicked "Sign off and committ suggestions" everywhere.

File: content/en/multitenancy.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+Multitenant software examples include Google Mail, Google Docs, Microsoft Office 365, Salesforce CRM, and Dropbox, among many more that categorize as either entirely or partially multitenant software. 
+
+A tenant can be as small as one independent user with a single login ID — think personal productivity software — or as large as an entire corporation with thousands of login IDs, each with its own privileges yet interrelated in multiple ways. 
+
+## Problem it addresses 
+
+Multitenancy is a key to operational efficiency of server-run software.  Providing each tenant with a dedicated installation is wasteful in terms of hardware utilization and maintenance efforts.  
+
+## How it helps
+
```
Comment (CathPag) [inline content/en/multitenancy.md:22]:
```suggestion

Multitenant software provides each tenant a segregated environment (work data, settings, list of credentials, etc.), simultaneously serving multiple tenants. From a tenant's perspective, each has its dedicated software installation, although, in reality, they are all sharing one. This is achieved by running the software on a server and allowing tenants to connect to it over the network via an interface and/or an [API](/application-programming-interface/) (also refer to [Client-Server Architecture](/client-server-architecture/)).
```
This was in the first section and fits very well in here

File: content/en/multitenancy.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+Multitenant software is designed in a way that tenants share the resources of one installation without affecting each other, or affecting only in predefined and controlled ways.  This results in huge savings for the software provider that can be passed on to the tenants, making services that would otherwise be very expensive much cheaper or even free (again, think web-based e-mail or document editors).
```
Comment (CathPag) [inline content/en/multitenancy.md:23]:
```suggestion
With multitenant software, tenants share the resources of one installation without affecting each other or only in predefined and controlled ways. The resulting resource savings on the software provider's side can be passed on to the tenants, significantly reducing the software cost for users (again, think web-based e-mail or document editors).
```
Same content with some edits

File: content/en/multitenancy.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+title: Multitenancy
+status: Feedback Appreciated
+category: Property
+---
+
+## What it is
+
+Multitenancy (or multi-tenancy) refers to a single software installation that serves multiple tenants. A tenant is a user, application, or a group of users/applications that utilize the software to operate on their own data set. These tenants don't share data (unless explicitly instructed by the owner) and may not even be aware of one another.  
+
+In other words, multitenant software provides each tenant with its segregated environment (work data, settings, list of credentials, etc.), simultaneously serving multiple tenants. From a tenant's perspective, each has its dedicated software installation, although, in reality, they are all sharing one. This is achieved by running the software on a server and allowing tenants to connect to it over the network via an interface and/or an [API](/application-programming-interface/) (also refer to [Client-Server Architecture](/client-server-architecture/)).
```
Comment (CathPag) [inline content/en/multitenancy.md:11]:
```suggestion

```
Moved this into the how it works section

File: content/en/multitenancy.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
```
Comment (CathPag) [inline content/en/multitenancy.md:12]:
```suggestion

A tenant can be as small as one independent user with a single login ID — think personal productivity software — or as large as an entire corporation with thousands of login IDs, each with its own privileges yet interrelated in multiple ways. 
```
I would put this before the example

File: content/en/multitenancy.md [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+Multitenant software examples include Google Mail, Google Docs, Microsoft Office 365, Salesforce CRM, and Dropbox, among many more that categorize as either entirely or partially multitenant software. 
+
+A tenant can be as small as one independent user with a single login ID — think personal productivity software — or as large as an entire corporation with thousands of login IDs, each with its own privileges yet interrelated in multiple ways.
```
Comment (CathPag) [inline content/en/multitenancy.md:15]:
```suggestion

```
Moved this up

File: content/en/multitenancy.md [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+Multitenant software examples include Google Mail, Google Docs, Microsoft Office 365, Salesforce CRM, and Dropbox, among many more that categorize as either entirely or partially multitenant software. 
+
+A tenant can be as small as one independent user with a single login ID — think personal productivity software — or as large as an entire corporation with thousands of login IDs, each with its own privileges yet interrelated in multiple ways. 
+
+## Problem it addresses 
+
+Multitenancy is a key to operational efficiency of server-run software.  Providing each tenant with a dedicated installation is wasteful in terms of hardware utilization and maintenance efforts.
```
Comment (CathPag) [inline content/en/multitenancy.md:19]:
```suggestion
Without multitenancy, each tenant would need a dedicated software installation. This increases resource utilization and maintenance efforts, ultimately software costs.
```

**Review by CathPag — COMMENTED** [review]
@sergeyborovoy, I moved a lot of things around. The first section contained parts that should go into the last section. Also some edits for the other two sections I had not reviewed yet.

**Comment by CathPag** [comment]
> Hi @CathPag, hi @seokho-son My apologies, I guess was under the impression that by clicking "Resolve conversation", I tell github to merge the suggested changes. Like I said, I'm not a regular user of github. I just unresolved everything and clicked "Sign off and committ suggestions" everywhere.

No worries. Glad to hear you were not totally ignoring my 😅

**Comment by CathPag** [comment]
@sergeyborovoy, just following up on this. I think it's pretty close! Would love to merge soon :)

**Comment by sergeyborovoy** [comment]
Hi @CathPag sorry about the delay, I've been traveling.  Just signed off on your changes, thank you for your input!

**Comment by CathPag** [comment]
Thanks, @sergeyborovoy! 

@iamNoah1, can you please review this PR?

**Review by iamNoah1 — APPROVED** [review]
Thanks @sergeyborovoy I like the content :)

File: content/en/multitenancy.md [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+
+## How it helps
+
+Multitenant software provides each tenant a segregated environment (work data, settings, list of credentials, etc.), simultaneously serving multiple tenants. From a tenant's perspective, each has its dedicated software installation, although, in reality, they are all sharing one. This is achieved by running the software on a server and allowing tenants to connect to it over the network via an interface and/or an [API](/application-programming-interface/) (also refer to [Client-Server Architecture](/client-server-architecture/)).
+With multitenant software, tenants share the resources of one installation without affecting each other or only in predefined and controlled ways. The resulting resource savings on the software provider's side can be passed on to the tenants, significantly reducing the software cost for users (again, think web-based e-mail or document editors).
+
+## Related terms
+
+Multitenancy is not synonymous with [SaaS](https://glossary.cncf.io/software-as-a-service), although it is very common for SaaS to be multitenant and even to feature multitenancy as one of its core benefits.
```
Comment (jihoon-seo) [inline content/en/multitenancy.md:27]:
```suggestion
Multitenancy is not synonymous with [SaaS](/software-as-a-service/), although it is very common for SaaS to be multitenant and even to feature multitenancy as one of its core benefits.
```

**Review by jihoon-seo — APPROVED** [review]
LGTM!

**Comment by CathPag** [comment]
Awesome, thanks for your contribution, @sergeyborovoy. Any chance I could interest you in helping with the [stateless app definition](https://github.com/cncf/glossary/issues/250)?

**Comment by sergeyborovoy** [comment]
Hi @CathPag, I was about to respond with an enthusiastic "yes" but then I went to the GitHub issue page and saw that this article seems to have an (unusually?) painful history.  I'd like to figure out why before I commit to it.  I'm traveling right now but some time before the next wekend I hope to be able to carve out a free moment for this.

Should I delete the sergeyborovoy:patch-1 branch of Multitenancy?

**Comment by CathPag** [comment]
It's just not an easy term, and we need someone with a strong technical background to work on it.  You seem to have one which is why I asked you :)

Yes, now that the term is merged, you can delete your branch.

**Pull request merged**